lib/Trash/TrashBackend.php

```diff
@@ -121,7 +121,7 @@ public function restoreItem(ITrashItem $item) {
 		if ($node === null) {
 			throw new NotFoundException();
 		}
-		if (!$this->userHasAccessToPath($item->getUser(), $folderId . '/' . $item->getOriginalLocation(), Constants::PERMISSION_UPDATE)) {
+		if (!$this->userHasAccessToPath($item->getUser(), $item->getPath(), Constants::PERMISSION_UPDATE)) {
```

Comment:
You sure this will still work?

Reply:
Yes since the ACLManager is now aware of trash folders